core/kazoo_documents/src/kzd_box_message.erl

```diff
@@ -120,23 +123,55 @@ new(AccountId, Props) ->
                               ),
     Description = props:get_value(<<"Description">>, Props, <<"voicemail message with media">>),
 
-    DocProps = [{<<"_id">>, MsgId}
+    DocProps = props:filter_undefined(
+               [{<<"_id">>, MsgId}
                ,{?KEY_NAME, Name}
                ,{?KEY_DESC, Description}
                ,{?KEY_SOURCE_TYPE, ?KEY_VOICEMAIL}
                ,{?KEY_SOURCE_ID, props:get_value(<<"Box-Id">>, Props)}
                ,{?KEY_MEDIA_SOURCE, <<"recording">>}
-               ,{?KEY_MEDIA_FILENAME, props:get_value(<<"Attachment-Name">>, Props)}
+               ,{?KEY_MEDIA_FILENAME, gen_attachment_filename(Props)}
+               ,{?KEY_TRANSCRIBE_FILENAME, gen_transcribe_filename(Props)}
                ,{?KEY_STREAMABLE, 'true'}
                ,{?KEY_UTC_SEC, UtcSeconds}
-               ],
+               ]),
     kz_doc:update_pvt_parameters(kz_json:from_list(DocProps)
                                 ,Db
                                 ,[{'type', type()}
                                  ,{'now', Timestamp}
                                  ]
                                 ).
 
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+-spec gen_attachment_filename(kz_term:proplist()) -> kz_term:ne_binary().
+gen_attachment_filename(Props) ->
+    case props:get_value(<<"Transcribe-Voicemail">>, Props)
+        andalso props:is_defined(<<"Required-Format">>, Props)
+        andalso props:get_value(<<"Required-Format">>, Props) =/= <<"wav">> of
```

Comment:
I think instead of checking if the key is defined and that the value is not `wav`, you could accomplish the same thing with `props:get_ne_binary_value(<<"Required-Format">>, Props, <<"wav">>) =/= <<"wav">>`. If the key doesn't exist, `<<"wav">>` is returned and the check is false.